### PR TITLE
Generalize Table and TableEntry models, NQLs and search Leaderboard

### PR DESCRIPTION
**Pull request by Raalsky**

### CODE REVIEW

File: neptune/new/internal/backends/nql.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    STRING_SET = "stringSet"
+    EXPERIMENT_STATE = "experimentState"
+
+
+@dataclass
+class NQLQueryAttribute(NQLQuery):
+    name: str
+    type: NQLAttributeType
+    operator: NQLAttributeOperator
+    value: str
```
Comment (pkasprzyk) [inline neptune/new/internal/backends/nql.py:74]: for current cases it's OK, but in general it's not `: str` (you can e.g. query using running_time)

**Review by pkasprzyk — COMMENTED** [review]
Nice code, some minor comments only.

But I think we should have a broader talk about NQL in client.

File: neptune/new/internal/backends/nql.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    AND = "AND"
+    OR = "OR"
+
+
+@dataclass
+class NQLQueryAggregate(NQLQuery):
+    items: Iterable[NQLQuery]
+    aggregator: NQLAggregator
+
+    def __repr__(self):
```
[the remark this answers is not in the document]
Comment (Raalsky) [reply neptune/new/internal/backends/nql.py]: Ok, let's go with `__str__` + `str()`

File: neptune/new/internal/backends/nql.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+from dataclasses import dataclass
+
+
+@dataclass
+class NQLQuery:
+    pass
+
+
+@dataclass
+class NQLEmpty:
```
[the remark this answers is not in the document]
Comment (Raalsky) [reply neptune/new/internal/backends/nql.py]: Done

File: neptune/new/internal/backends/hosted_neptune_backend.py [hunk 4]
```diff
@@ -1089,28 +1090,23 @@ def _get_file_set_download_request(
...
-                    sortBy=["shortId"],
-                    sortFieldType=["string"],
-                    sortDirection=["ascending"],
-                    limit=limit,
-                    offset=offset,
+                    type=list(
+                        map(lambda container_type: container_type.to_api(), types)
+                    ),
+                    params={
+                        "query": {"query": str(query)},
```
[the remark this answers is not in the document]
Comment (Raalsky) [reply neptune/new/internal/backends/hosted_neptune_backend.py]: I think it's done :wink:

File: neptune/new/internal/backends/nql.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (pkasprzyk) [reply neptune/new/internal/backends/nql.py]: I don't think you use it anymore, but no harm done I guess.

**Review by pkasprzyk — APPROVED** [review]

**Review by pkasprzyk — APPROVED** [review]

**Pull request merged**